cluster/src/main/java/io/scalecube/cluster/metadata/MetadataStore.java

```diff
@@ -65,4 +71,5 @@
    * @return old metadata or null
    */
   Map<String, String> removeMetadata(Member member);
+
```

Comment:
Let's not have constant `String INCARNATION_KEY = "incarnation"` and let's not build logic around special key in hashmap. Let's define a new class, call it `Metadata`  with two fields `int version` and `Map<String, String> properties`. Let's take as fact that our abstraction of metadata became little bit more complex than plain Map, now it has another property -- version, so those two guys deserves an own class.